File: javascript/src/Behavioral/Interpreter/DvdActorTitleExpression.js
(function() {
  var __hasProp = {}.hasOwnProperty,
    __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

  define(['Behavioral/Interpreter/DvdAbstractExpression'], function(DvdAbstractExpression) {
    var DvdActorTitleExpression;
    return DvdActorTitleExpression = (function(_super) {
      __extends(DvdActorTitleExpression, _super);

      function DvdActorTitleExpression(title) {
        this.title = title;
      }

      DvdActorTitleExpression.prototype.interpret = function(ctx) {
        return ctx.getActorsForTitle(this.title).join(', ');
      };

      return DvdActorTitleExpression;

    })(DvdAbstractExpression);
  });

}).call(this);

File: go/behavioral/interpreter/DvdActorTitleExpression.go

package interpreter

import (
	"strings"
)

type DvdActorTitleExpression struct { // implements Interpretable.go
   actor string
}

func NewDvdActorTitleExpression(actor string) *DvdActorTitleExpression {
	return &DvdActorTitleExpression{ actor }
}
    
func (this *DvdActorTitleExpression) Interpret (ctx *DvdInterpreterContext) string {
	if this.actor == "" {
		return ""
	}
	return strings.Join(ctx.GetActorsForTitle( this.actor ),", ")
}
